src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -566,6 +508,10 @@ public static SSTableReader open(Descriptor descriptor,
 
             return sstable;
         }
+        catch (UnknownColumnException e)
+        {
+            throw new IllegalStateException(e);
+        }
         catch (Throwable t)
         {
             sstable.selfRef().release();
```

Comment:
@jasonstack Can `sstable` be null here if we throw an exception from `build()` that isn't `UnknownColumnException`?

Reply:
split into two try-catch..